pkg/dataset/config.go

```diff
@@ -72,12 +72,12 @@ func (c *Config) Apply(conn *pgx.Conn) error {
 	log.Info("msg", fmt.Sprintf("Setting trace dataset default retention period to %s", c.Traces.RetentionPeriod))
 
 	queries := map[string]interface{}{
-		setDefaultMetricChunkIntervalSQL:    time.Duration(c.Metrics.ChunkInterval),
+		setDefaultMetricChunkIntervalSQL:    time.Duration(c.Metrics.ChunkInterval).String(),
 		setDefaultMetricCompressionSQL:      c.Metrics.Compression,
-		setDefaultMetricHAReleaseRefreshSQL: time.Duration(c.Metrics.HALeaseRefresh),
-		setDefaultMetricHAReleaseTimeoutSQL: time.Duration(c.Metrics.HALeaseTimeout),
-		setDefaultMetricRetentionPeriodSQL:  time.Duration(c.Metrics.RetentionPeriod),
-		setDefaultTraceRetentionPeriodSQL:   time.Duration(c.Traces.RetentionPeriod),
+		setDefaultMetricHAReleaseRefreshSQL: time.Duration(c.Metrics.HALeaseRefresh).String(),
```

Comment:
I am a bit surprised with this. How is `go` duration string working with PgSQL interval?

Reply:
+1

Reply:
It works because this is valid in Postgres

```
promscale@/tmp:promscale> select '8h0m0s'::interval;
+----------+
| interval |
|----------|
| 8:00:00  |
+----------+
```

But I double checked, and just the Lease values needs to be converted to string because the SQL does `$1::text`. The way pgx works is that it has the text OID registered to a Codec, that Codex expects the argument to be a `string`, `[]byte` or a TextValuer interface, since time.Duration is none of those it errors.

I'm updating this part and removing the `.String()` where it's not needed.

Reply:
well, that's handy